src/librustc_mir/interpret/intrinsics.rs

```diff
@@ -399,6 +399,11 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 );
                 self.copy_op(self.operand_index(args[0], index)?, dest)?;
             }
+            sym::likely | sym::unlikely => {
+                // These just return their argument
+                let a = self.read_immediate(args[0])?;
+                self.write_immediate(*a, dest)?;
```

Comment:
I believe

```suggestion
                self.copy_op(args[0], dest)?;
```

will do the same thing, but with more checks.

Reply:
I think the amount of checking is the same... except the `immediate` code also requires that the value be an immediate, whereas `copy_op` also works with non-immediate values.

I do not have a strong preference.